Rewrite context and make various fixes

Koochr opened the pull request.

dsm-w3f commented:
@Koochr I tested the version and the only problem that I found was when I deleted the post I cannot see it in the trash. Other things are all working. The code is ok. Just remember that we have PRs #16, #17 and #14 open, which fix bugs and add features. Maybe would be better to merge then before this PR. Otherwise the modifications could be difficult to integrate.

michelnapo reviewed (changes requested):
@Koochr I tested the version and the only problem that I found was when I deleted the post I cannot see it in the trash. Other things are all working. The code is ok. Just remember that we have PRs #16, #17 and #14 open, which fix bugs and add features. Maybe would be better to merge then before this PR. Otherwise the modifications could be difficult to integrate.

Koochr commented:
> @Koochr I tested the version and the only problem that I found was when I deleted the post I cannot see it in the trash. Other things are all working. The code is ok. Just remember that we have PRs #16, #17 and #14 open, which fix bugs and add features. Maybe would be better to merge then before this PR. Otherwise the modifications could be difficult to integrate.

Thanks, I've fixed.
Regarding other PRs - I've rewritten all the context, so it let's do it in reverse order, first merge this one, and then I'll include https://github.com/pointnetwork/blogsoftware.point/pull/16 and https://github.com/pointnetwork/blogsoftware.point/pull/14 manually.
Regarding https://github.com/pointnetwork/blogsoftware.point/pull/17 - I don't understand, what sense does it make? And how is it connected with deleted blogs?
